core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartds/StopStartDownscaleActions.java

```diff
@@ -91,15 +92,39 @@ protected void prepareExecution(StopStartDownscaleTriggerEvent payload, Map<Obje
             }
 
             @Override
-            protected void doExecute(
-                    StopStartDownscaleContext context, StopStartDownscaleTriggerEvent payload, Map<Object, Object> variables) throws Exception {
-                stopStartDownscaleFlowService.clusterDownscaleStarted(context.getStack().getId(), payload.getHostGroup(), payload.getHostIds());
-                sendEvent(context);
+            protected void doExecute(StopStartDownscaleContext context, StopStartDownscaleTriggerEvent payload, Map<Object, Object> variables) throws Exception {
+                StackDtoDelegate stack = context.getStack();
+                stopStartDownscaleFlowService.initScaleDown(stack.getId(), payload.getHostGroup());
+                if (payload.isFailureRecoveryEnabled()) {
+                    LOGGER.info("Identifying recovery candidates for downscale");
```

Comment:
Nit: Put all the following lookups into this same if condition, and generate empty lists otherwise ?

Reply:
Updated this